zerver/tests/test_message_fetch.py

```diff
@@ -3442,6 +3444,164 @@ def test_parse_anchor_value(self) -> None:
+        search_date_str = search_datetime.isoformat()
+        result = get_message_search_result(search_date_str)
+        self.assertEqual(result["anchor"], message_ids[-1])
+        self.assertEqual(result["found_newest"], True)
+        self.assertEqual(result["found_oldest"], False)
+        verify_message_ids_and_closest_message_id(result, search_datetime)
+
+        # Case: Invalid search date format
+        with self.assertRaises(ValidationError) as validation_error:
+            search_date_str = search_datetime.strftime("%d-%b-%Y")
+            get_message_search_result(search_date_str)
+        self.assertEqual(
+            validation_error.exception.message,
+            "Search date must be in the ISO 8601 format.",
+        )
+
+        # Case: Missing 'anchor_date' argument
+        with self.assertRaises(JsonableError) as error:
+            result = get_message_search_result(search_date="")
+        self.assertEqual(
+            str(error.exception),
+            "Missing 'anchor_date' argument.",
+        )
+
+        # Case: No messages found for the narrow conditions even though a message is sent on the `search_datetime`
+        search_date_str = search_datetime.isoformat()
+        query_params = dict(
+            anchor="date",
+            anchor_date=search_date_str,
+            num_before=5,
+            num_after=5,
+            narrow=f'[["stream", "{stream.name}"], ["search", "thisissomerandomsearch"]]',
+        )
+        request = HostRequestMock(query_params, user_profile)
+        payload = get_messages_backend(request, user_profile)
+        result = orjson.loads(payload.content)
+        self.assertEqual(result["anchor"], LARGER_THAN_MAX_MESSAGE_ID)
+        self.assertEqual(result["found_newest"], True)
+        self.assertEqual(result["found_oldest"], True)
+
+        # Case: Anchor should point to the first message when multiple messages are sent on a single day
+        first_message_id = None
+        current_datetime += datetime.timedelta(days=search_by_days * 50)
+        with time_machine.travel(current_datetime) as traveller:
+            for _ in range(20):
+                message_id = self.send_stream_message(user_profile, stream.name)
+                if first_message_id is None:
+                    first_message_id = message_id
+
+        search_datetime = current_datetime
+        search_date_str = search_datetime.isoformat()
+        result = get_message_search_result(search_date_str)
+        self.assertEqual(result["anchor"], first_message_id)
+        self.assertEqual(result["found_newest"], False)
+        self.assertEqual(result["found_oldest"], False)
+
+        # Case: When only date filter is passed.
+        search_datetime = datetime.datetime(1888, 6, 2, tzinfo=datetime.timezone.utc)
+        search_date_str = search_datetime.isoformat()
+        result = get_message_search_result(search_date_str)
+        self.assertEqual(result["anchor"], 151)
```

Comment:
can someone help me - what's the easiest way to fetch the message ID of the first message sent? I'll replace it with `151`.

Reply:
I'm not sure what you're getting at -- `self.send_stream_message` returns the message ID of a message you sent. 

Is the issue you want the very oldest message available to a user? One option would be to just do a `GET /messages` query with `anchor: "oldest"` without a date filter, and verify the results agree.

Reply:
> Is the issue you want the very oldest message available to a user? 

yes; thanks for the suggestion.